app/services/katello/pulp3/content_view_version/import.rb

```diff
@@ -45,14 +45,67 @@ def destroy_importer(importer_href)
-          def check_permissions!(path)
+          def check!(content_view:, metadata:, path:)
+            check_permissions!(metadata: metadata, path: path)
+            check_metadata!(content_view: content_view, metadata: metadata)
+          end
+
+          # rubocop:disable Metrics/MethodLength
+          # rubocop:disable Metrics/AbcSize
+          def check_metadata!(content_view:, metadata:)
+            major = metadata[:content_view_version][:major]
+            minor = metadata[:content_view_version][:minor]
+
+            if ::Katello::ContentViewVersion.where(major: major, minor: minor, content_view: content_view).exists?
+              fail _("Content View Version specified in the metadata - '%{name}' already exists. "\
+                      "If you wish to replace the existing version, delete %{name} and try again. " % { name: "#{content_view.name} #{major}.#{minor}" })
+            end
+
+            if metadata[:from_content_view_version].present?
+              from_major = metadata[:from_content_view_version][:major]
+              from_minor = metadata[:from_content_view_version][:minor]
+
+              unless ::Katello::ContentViewVersion.where(major: from_major, minor: from_minor, content_view: content_view).exists?
+                fail _("Prior Content View Version specified in the metadata - '%{name}' does not exist. "\
+                        "Please import the metadata for '%{name}' before importing '%{current}' " % { name: "#{content_view.name} #{from_major}.#{from_minor}",
+                                                                                                      current: "#{content_view.name} #{major}.#{minor}"})
+              end
+            end
+
+            repos_from_data = metadata[:repository_mapping].values.map do |value|
+              ::Katello::Repository.joins(:root, :product).where("#{::Katello::Product.table_name}" => {:name => value[:product]},
+                                                                 "#{::Katello::RootRepository.table_name}" => {:name => value[:repository]})
+            end
+
+            matching_repos = content_view.repositories.where("#{::Katello::Repository.table_name}.id" => repos_from_data.inject(&:or))
+            if metadata[:repository_mapping].count != content_view.repositories.count || matching_repos.count != content_view.repositories.count
+              repos_in_content_view = content_view.repositories.map do |repo|
+                _("\n* Product = '%{product}', Repository = '%{repository}'" % { product: repo.product.name, repository: repo.name })
+              end
+
+              repos_in_import = metadata[:repository_mapping].values.map do |value|
+                _("\n* Product = '%{product}', Repository = '%{repository}'" % { product: value[:product], repository: value[:repository] })
+              end
+
+              fail _("Repositories in the importing content view do not match the repositories provide in the import metadata.\n "\
```

Comment:
```suggestion
              fail _("Repositories in the importing content view do not match the repositories provided in the import metadata.\n "\
```
"Importing content view" isn't self-explanatory.  Does this mean the content view I'm importing from, or to?